ArcGISRuntimeSDKQt_CppSamples/Analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.cpp

```diff
@@ -0,0 +1,353 @@
+// [WriteFile Name=ConfigureSubnetworkTrace, Category=Analysis]
+// [Legal]
+// Copyright 2020 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "ConfigureSubnetworkTrace.h"
+
+#include "CodedValueDomain.h"
+#include "TaskWatcher.h"
+#include "UtilityAssetGroup.h"
+#include "UtilityAssetType.h"
+#include "UtilityCategory.h"
+#include "UtilityCategoryComparison.h"
+#include "UtilityDomainNetwork.h"
+#include "UtilityElement.h"
+#include "UtilityElementTraceResult.h"
+#include "UtilityNetwork.h"
+#include "UtilityNetworkDefinition.h"
+#include "UtilityNetworkSource.h"
+#include "UtilityNetworkTypes.h"
+#include "UtilityTerminal.h"
+#include "UtilityTerminalConfiguration.h"
+#include "UtilityTier.h"
+#include "UtilityTraceAndCondition.h"
+#include "UtilityTraceConfiguration.h"
+#include "UtilityTraceOrCondition.h"
+#include "UtilityTraceParameters.h"
+#include "UtilityTraceResultListModel.h"
+#include "UtilityTraversability.h"
+
+#include <QQmlEngine>
+#include <algorithm>
+
+using namespace Esri::ArcGISRuntime;
+QString expressionToString(UtilityTraceConditionalExpression* expression);
+QString comparisonOperatorToString(const UtilityAttributeComparisonOperator& comparisonOperator);
+QVariant convertToDataType(const QVariant& value, const Esri::ArcGISRuntime::UtilityNetworkAttributeDataType& dataType);
```

Comment:
My mistake, I meant these should remain class functions but become static class functions. They all seemed like helpers that don't need access to any class members.